client/containers/Pub/PubSyncManager.tsx

```diff
@@ -165,7 +190,9 @@ class PubSyncManager extends React.Component<Props, State> {
 			firebaseDraftRef: null,
 			pubData: this.props.pubData,
 			collabData: {
-				editorChangeObject: {},
+				// TODO(ian): Verify that there are no unchecked property accesses on this
+				// editorChangeObject and then remove this cast.
+				editorChangeObject: {} as unknown as null,
```

Comment:
This is a bit of paranoia on my part.

The gist is this: during the first couple of renders, there is no `EditorChangeObject`, because ProseMirror has to set that up and then bubble it into React. So the sensible way to represent that is to initialize it to `null` and give it type `null | EditorChangeObject`. But...there is a lot of old code that relies on the old fallback of `{}` as the initial value, and I'm not totally sure that I addressed it all. E.g., the expression:

```ts
collabData.editorChangeObject.view
```

is safe for `editorChangeObject = {}` but not for `editorChangeObject = null`. So in effect we're telling the compiler about the state we'd eventually like to get to (`null | EditorChangeObject`) while also giving the runtime code a way to not crash if part of it is still trying to access an object instead of `null` here.

I digress, but actually this would be a great use for Proxies. You could write something like this:

```ts
const accessWarningProxy = new Proxy({} , {
    get: (_, name) => console.warn(`Tried to access property ${name} on supposedly null EditorChangeObject!`),
});
```

and throw that in there in place of `{}`. Over time we'd see console warnings when some piece of the code tries to do property access when `editorChangeObject` is supposed to be null, and once we stop seeing them for a while, we could remove the cast.

Reply:
Double teachable moment. Thanks for breaking that down two times.